File: DevExpress-Demo/VB/ControlsDemo.Wpf/ViewModels/NewJumpTaskWindowViewModel.vb
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
Imports System.Windows
Imports CommonDemo.Helpers
Imports DevExpress.Mvvm.DataAnnotations

Namespace ControlsDemo

    <POCOViewModel(ImplementIDataErrorInfo:=True)>
    Public Class NewJumpTaskWindowViewModel

        Public Sub New()
            Icons = New NamedIcon() {New NamedIcon() With {.Caption = "Moon", .Icon = ImagesHelper.GetSvgImage("pack://application:,,,/ControlsDemo;component/Images/Moon.svg", New Size(16, 16))}, New NamedIcon() With {.Caption = "Sun", .Icon = ImagesHelper.GetSvgImage("pack://application:,,,/ControlsDemo;component/Images/Sun.svg", New Size(16, 16))}}
            CustomCategory = ""
            Title = "Title"
            Description = ""
            MessageText = "Message"
            Icon = Icons.ElementAt(0)
        End Sub

        Public Property Icons As IEnumerable(Of NamedIcon)

        Public Overridable Property CustomCategory As String

        <Required>
        Public Overridable Property Title As String

        Public Overridable Property Icon As NamedIcon

        Public Overridable Property Description As String

        Public Overridable Property MessageText As String
    End Class
End Namespace

File: DevExpress-Demo/CS/ControlsDemo.Wpf/ViewModels/NewJumpTaskWindowViewModel.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Windows;
using CommonDemo.Helpers;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Utils;
using DevExpress.Xpf.Core.Native;

namespace ControlsDemo {
    [POCOViewModel(ImplementIDataErrorInfo = true)]
    public class NewJumpTaskWindowViewModel {
        public NewJumpTaskWindowViewModel() {
            Icons = new NamedIcon[] {
                new NamedIcon() {
                    Caption = "Moon",
                    Icon = ImagesHelper.GetSvgImage("pack://application:,,,/ControlsDemo;component/Images/Moon.svg", new Size(16,16)) },
                new NamedIcon() {
                    Caption = "Sun",
                    Icon = ImagesHelper.GetSvgImage("pack://application:,,,/ControlsDemo;component/Images/Sun.svg", new Size(16,16)) }
            };
            CustomCategory = "";
            Title = "Title";
            Description = "";
            MessageText = "Message";
            Icon = Icons.ElementAt(0);
        }
        public IEnumerable<NamedIcon> Icons { get; set; }
        public virtual string CustomCategory { get; set; }
        [Required]
        public virtual string Title { get; set; }
        public virtual NamedIcon Icon { get; set; }
        public virtual string Description { get; set; }
        public virtual string MessageText { get; set; }
    }
}
